### PR TITLE
Propagate harmonic interpolation method option to Navier Stokes

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #21923

No regolds! Our viscosities must be constant over all of NS. This will change with K-eps

This does not touch the mixing length. I dont see a big need to improve it as we ll move to regularized K-epsilon soon

### CODE REVIEW

File: framework/include/utils/MathFVUtils.h [hunk 1]
```diff
@@ -175,6 +175,14 @@ enum class InterpMethod
   QUICK
 };
 
+/*
+ * Converts from the interpolation method to the interpolation enum.
+ * This routine is here in lieu of using a MooseEnum for InterpMethod
+ * @param interp_method the name of the interpolation method
+ * @return the interpolation method
+ */
+InterpMethod selectInterpolationMethod(std::string interp_method);
```
Comment (lindsayad) [inline framework/include/utils/MathFVUtils.h:184]:
```suggestion
InterpMethod selectInterpolationMethod(const std::string & interp_method);
```

File: framework/src/utils/MathFVUtils.C [hunk 2]
```diff
@@ -84,6 +84,33 @@ onBoundary(const std::set<SubdomainID> & subs, const FaceInfo & fi)
   }
 }
 
+InterpMethod
+selectInterpolationMethod(std::string interp_method)
```
Comment (lindsayad) [inline framework/src/utils/MathFVUtils.C:88]:
```suggestion
selectInterpolationMethod(const std::string & interp_method)
```

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusion.C [hunk 3]
```diff
@@ -38,9 +46,14 @@ INSFVMomentumDiffusion::INSFVMomentumDiffusion(const InputParameters & params)
...
-  const auto face = Moose::FV::makeCDFace(*_face_info, faceArgSubdomains());
   const auto dudn = gradUDotNormal();
-  const auto face_mu = _mu(face);
+  ADReal face_mu;
+  Moose::FV::interpolate(_mu_interp_method,
+                         face_mu,
+                         _mu(elemFromFace()),
+                         _mu(neighborFromFace()),
+                         *_face_info,
+                         true);
```
Comment (lindsayad) [inline modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusion.C:56]: My heart prefers that we perhaps make another `makeFace` overload and then keep the call to the functor with the `FaceArg`.

But my head says this is a shorter function call stack and consequently maybe faster.

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusion.C (hunk 3 above)
Comment (GiudGiud) [reply modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusion.C:56]: the cool thing with writing it like this is that it looks the same for k and eps * k for example

**Review by lindsayad — APPROVED** [review]
It would be nice to unit test the missing coverage in the framework

**Comment by GiudGiud** [comment]
for sure. At least before we add the vector limiter options

**Pull request merged**